presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -298,40 +302,44 @@ public synchronized void setRevocableMemoryReservation(long newRevocableMemoryRe
     {
         checkArgument(newRevocableMemoryReservation >= 0, "newRevocableMemoryReservation is negative");
 
-        long delta = newRevocableMemoryReservation - revocableMemoryReservation;
+        long delta = newRevocableMemoryReservation - getReservedRevocableBytes();
 
         if (delta > 0) {
             reserveRevocableMemory(delta);
         }
         else {
-            freeRevocableMemory(-delta);
+            operatorMemoryContext.updateRevocableMemory(delta);
         }
     }
 
-    public synchronized void freeRevocableMemory(long bytes)
+    public void freeMemory(long bytes)
     {
         checkArgument(bytes >= 0, "bytes is negative");
-        checkArgument(bytes <= revocableMemoryReservation, "tried to free more revocable memory than is reserved");
-        driverContext.freeRevocableMemory(bytes);
-        revocableMemoryReservation -= bytes;
+        operatorMemoryContext.updateUserMemory(-bytes);
     }
 
-    public void freeMemory(long bytes)
+    public void freeAllMemory()
```

Comment:
I prefer that a word different from `free` is used. Maybe `close`, `destroy`, `clear`?

It would be ideal if the function names expresses the intent that: this is the end, all memory needs to be freed, and no more memory will be reserved.